unblob/handlers/archive/instar/hd.py

```diff
@@ -0,0 +1,105 @@
+import io
+import struct
+from pathlib import Path
+from typing import Optional
+
+from structlog import get_logger
+
+from unblob.file_utils import InvalidInputFormat, iterate_patterns
+from unblob.models import Extractor, File, HexString, StructHandler, ValidChunk
+
+logger = get_logger()
+
+EOCD_RECORD_HEADER = 0x9054B50
+
+
+class InstarHDExtractor(Extractor):
+    def extract(self, inpath: Path, outpath: Path):
+        with open(inpath, "rb") as file:
+            outfile = outpath.joinpath(f"{inpath.name}")
+            with open(outfile, "wb") as out_file:
+                headers = file.read()
+
+                if b"\x50\x4b\x03\x07" in headers:
+                    headers = headers.replace(b"\x50\x4b\x03\x07", b"\x50\x4b\x03\x04")
+
+                if b"\x50\x4b\x05\x09" in headers:
+                    headers = headers.replace(b"\x50\x4b\x05\x09", b"\x50\x4b\x05\x06")
+
+                if b"\x50\x4b\x01\x08" in headers:
+                    headers = headers.replace(b"\x50\x4b\x01\x08", b"\x50\x4b\x01\x02")
+                out_file.write(headers)
```

Comment:
shouldn't we also run the zip extraction here? This is like the duplicate of the zip handler, except a few bytes are replaced in the carved out content?

Reply:
Two ways of going about it:
1. we carve the chunk, our extractor converts it to classic zip, zip handler picks it up
2. we carve the chunk, our extractor converts it to classic zip **in place** and calls the zip extractor on it using a `Command`

While option 2 is more efficient, it kinda breaks the contract that a carved out chunk corresponds exactly to what was found in its parent file from start offset to end offset.